src/mmds/src/token_headers.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use micro_http::{HttpHeaderError, RequestError};
+use std::collections::HashMap;
+use std::fmt;
+use std::fmt::{Display, Formatter};
+use std::result::Result;
+
+/// Wrapper over supported HTTP Custom Header types.
+#[derive(Debug, Eq, Hash, PartialEq)]
+pub enum TokenHeader {
+    /// Header `X-metadata-token`
+    XMetadataToken,
+    /// Header `X-metadata-token-ttl-seconds`
+    XMetadataTokenTtlSeconds,
+}
+
+impl Display for TokenHeader {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match *self {
+            TokenHeader::XMetadataToken => write!(f, "X-metadata-token"),
+            TokenHeader::XMetadataTokenTtlSeconds => {
+                write!(f, "X-metadata-token-ttl-seconds")
+            }
+        }
+    }
+}
```

Comment:
We could also define these values as constants associated with the `TokenHeaders` struct:
```
impl TokenHeaders {
    const X_METADATA_TOKEN: &'static str = "X-metadata-token";
    const X_METADATA_TOKEN_TTL_SECONDS: &'static str = "X-metadata-token-ttl-seconds";
    ...
}
```

Not sure if it's necessarily better, but I have a slight preference towards this option.

Reply:
I think it's a good idea. Done.